### DISCUSSION TITLE
Can I run Python Code from an Arduino C firmware?

### DISCUSSION THREAD

**Discussion by hacktronics** [micropython/micropython#10264] [General]
I have some Arduino based C code, I want to add feature to support some python scripts received over WiFi, it is possible.
Is there some embeddable python parser, than can be used?
Or I have to add micro python firmware, and need a way to switch to the python firmware at runtime.
Any ideas are welcome, on how to solve the same.

**Comment by jimmo** [comment]
Yes this is possible -- you can embed MicroPython into any C or C++ codebase. However to get access to hardware functionality you will need to implement the relevant bindings. But if your goal is just to provide a way to run custom scripts with an API you control (in addition to standard Python built-in modules) then this is doable.

See https://github.com/micropython/micropython/tree/master/examples/embedding however we are in the process of reworking this, so https://github.com/micropython/micropython/pull/9529 is the latest.

**Reply by hacktronics** [reply]
Is there an Arduino example? There is an makefile system for Arduino, does simply including files will work. Where to provide own API implementation in that?

**Reply by jimmo** [reply]
@hacktronics Sorry about the delay responding.

There isn't an example, other than the two that I already linked to.
